### PR TITLE
Reset password validation code improved

### LINKED ISSUE #548: Improve reset password validation code
**Actual Behaviour**

* Reset password form validation code can be improved with switch statement.
* If there is an error in one field, and we start typing in other field, the error in the previous field is not displayed.
* After making the 'New Password' and 'Confirm Password' fields equal, if we make any changes in 'New Password', the confirm password error of unequal passwords is not shown.

**Expected Behaviour**

* Above stated problems should not occur

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by rossoskull**
Fixes #548 

Changes: 
* Reset password validation code made better
* If there is an error in one field, and we start typing in other field, the error in the previous field is displayed.
* After making the 'New Password' and 'Confirm Password' fields equal, if we make any changes in 'New Password', the confirm password error of unequal passwords is shown.

Surge Deployment Link: https://pr-549-fossasia-susi-accounts.surge.sh

### CODE REVIEW

**Comment by samagragupta** [comment]
@fragm3 Sorry but an issue is made on this : https://github.com/fossasia/accounts.susi.ai/issues/551

**Comment by rossoskull** [comment]
@fragm3 looks like I made a mistake in naming the issue and PR, the changes are for reset password form :sweat_smile:

**Comment by rossoskull** [comment]
Changes made, please review :smile:

**Comment by anshumanv** [comment]
Good work here. :tada: 

You may want to keep the `Save Changes` button disabled till the form is valid.

File: src/components/Auth/ChangePassword/ChangePassword.react.js [hunk 1]
```diff
@@ -106,75 +106,63 @@ export default class ChangePassword extends Component {
...
-      !this.state.confirmPasswordError
-    ) {
-      state.validForm = true;
-    } else {
-      state.validForm = false;
+    // eslint-disable-next-line
+    switch (event.target.name) {
+      case 'currentPassword':
+        state.currentPasswordError = !(state.currentPassword.length >= 6);
+        if (state.currentPasswordError) {
```
Comment (akshatnitd) [inline src/components/Auth/ChangePassword/ChangePassword.react.js]: Use ternary here, so that we can do it in less lines of code.

**Comment by rossoskull** [comment]
Done @akshatnitd @anshumanv 
ezgif-1-51aff6336441

**Pull request merged**